File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/examples/scoreboard/ScoreboardExample.java
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers.examples.scoreboard;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase;
import com.gmail.borlandlp.minigamesdtools.arena.gui.providers.GUIProvider;
import com.gmail.borlandlp.minigamesdtools.arena.team.TeamProvider;
import me.johnnykpl.scoreboardwrapper.ScoreboardWrapper;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class ScoreboardExample extends GUIProvider {
    private ScoreboardWrapper scoreboardWrapper;

    public ScoreboardExample(ArenaBase arena) {
        super(arena);
    }

    public ScoreboardExample() {}

    @Override
    public void onInit() {
        this.scoreboardWrapper = new ScoreboardWrapper("Title");
        this.scoreboardWrapper.addLine("example text");
        this.scoreboardWrapper.addBlankSpace();
        this.scoreboardWrapper.addLine("example text 2");
    }

    @Override
    public void beforeGameStarting() {

    }

    @Override
    public void gameEnded() {
        for(TeamProvider team : this.getArena().getTeamController().getTeams()) {
            for (Player player : team.getPlayers()) {
                if (player != null) {
                    player.setScoreboard(Bukkit.getScoreboardManager().getNewScoreboard());
                }
            }
        }
    }

    @Override
    public void update() {
        for(TeamProvider team : this.getArena().getTeamController().getTeams()) {
            for (Player player : team.getPlayers()) {
                if (player != null) {
                    player.setScoreboard(this.scoreboardWrapper.getScoreboard());
                }
            }
        }
    }

    @Override
    public void beforeRoundStarting() {

    }

    @Override
    public void onRoundEnd() {

    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/examples/scoreboard/ScoreboardExample.kt
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers.examples.scoreboard

import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase
import com.gmail.borlandlp.minigamesdtools.arena.gui.providers.GUIProvider
import me.johnnykpl.scoreboardwrapper.ScoreboardWrapper
import org.bukkit.Bukkit

class ScoreboardExample : GUIProvider {
    private var scoreboardWrapper: ScoreboardWrapper? = null

    constructor(arena: ArenaBase?) : super(arena) {}
    constructor() {}

    override fun onInit() {
        scoreboardWrapper = ScoreboardWrapper("Title")
        scoreboardWrapper!!.addLine("example text")
        scoreboardWrapper!!.addBlankSpace()
        scoreboardWrapper!!.addLine("example text 2")
    }

    override fun beforeGameStarting() {}
    override fun gameEnded() {
        for (team in arena!!.teamController!!.teams) {
            for (player in team.players) {
                player.scoreboard = Bukkit.getScoreboardManager().newScoreboard
            }
        }
    }

    override fun update() {
        for (team in arena!!.teamController!!.teams) {
            for (player in team.players) {
                player.scoreboard = scoreboardWrapper!!.scoreboard
            }
        }
    }

    override fun beforeRoundStarting() {}
    override fun onRoundEnd() {}
}